logstash-core/lib/logstash/webserver.rb

```diff
@@ -59,40 +56,34 @@ def error(str)
       logger.error(str)
     end
 
+    def rack_app
+      namespaces = rack_namespaces # These need this scope
+      Rack::Builder.new do
+        run LogStash::Api::Modules::Root
+        namespaces.each_pair do |namespace, app|
+          map(namespace) do
+            run app
+          end
+        end
+      end
+    end
+
+    def rack_namespaces
```

Comment:
why should the webserver class care about the resources (web endpoints) defined in the webapp? for me having them here complicate the separation of concerns.